src/backend/distributed/commands/utility_hook.c

```diff
@@ -255,9 +256,14 @@ citus_ProcessUtility(PlannedStmt *pstmt,
 
 	if (!CitusHasBeenLoaded())
 	{
-		if (!IsMainDB)
+		if (!IsMainDB &&
+			!IsCommandToCreateMainDB(parsetree))
 		{
 			RunPreprocessMainDBCommand(parsetree, queryString);
+			if (IsA(parsetree, CreatedbStmt))
```

Comment:
you probably want to do something similar for DROP DATABASE

Reply:
I added DROP DB too, thanks 😄